core/src/execution/sort.rs

```diff
@@ -159,12 +159,16 @@ where
                     pos += 1;
                 }
             } else {
-                unsafe {
-                    ptr::copy_nonoverlapping(
-                        bucket.as_ptr(),
-                        self.get_unchecked_mut(pos),
-                        bucket.len(),
-                    );
+                if !bucket.is_empty() {
```

Comment:
Without this check we were hitting a debug assertion on the call to `get_unchecked_mut(pos)`

Reply:
what will happen if this check doesn't satisfy? 🤔

Reply:
nvm, if its empty, we should just skip it